lib/pure/os.nim

```diff
@@ -3194,6 +3208,30 @@ proc setLastModificationTime*(file: string, t: times.Time) {.noNimScript.} =
     discard h.closeHandle
     if res == 0'i32: raiseOSError(osLastError(), file)
 
+func isValidFilename*(path: string, maxLen = 259.Positive): bool {.since: (1, 1).} =
+  ## Returns true if ``path`` filename is valid for crossplatform use.
+  ## This is useful if you want to copy or save files across Windows, Linux, Mac, etc.
+  ## Uses ``invalidFilenameChars``, ``invalidFilenames`` and ``maxLen`` for checking.
+  ##
+  ## .. code-block:: nim
+  ##   assert not isValidFilename(" foo")    ## Leading white space
+  ##   assert not isValidFilename("foo ")    ## Trailing white space
+  ##   assert not isValidFilename("foo.")    ## Ends with Dot
+  ##   assert not isValidFilename("con.txt") ## "CON" is invalid (Windows)
+  ##   assert not isValidFilename("OwO:UwU") ## ":" is invalid (Mac)
+  ##   assert not isValidFilename("aux.bat") ## "AUX" is invalid (Windows)
+  ##
+  # https://docs.microsoft.com/en-us/dotnet/api/system.io.pathtoolongexception
+  # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx
```

Comment:
IIRC there is a docgen bug which will mean that these are included in the doc comment, might have been fixed by now, but please check.

Reply:
`nim doc` works Ok. _"Works on my machine"_